app/[locale]/10years/_components/TorchHistorySwiper/index.tsx

```diff
@@ -19,6 +19,22 @@ type TorchHistorySwiperProps = {
   currentHolderAddress: Address | null
 }
 
+const getOrdinalSuffix = (num: number): string => {
+  const lastDigit = num % 10;
+  const lastTwoDigits = num % 100;
+  
+  if (lastTwoDigits >= 11 && lastTwoDigits <= 13) {
+    return `${num}th`;
+  }
+  
+  switch (lastDigit) {
+    case 1: return `${num}st`;
+    case 2: return `${num}nd`;
+    case 3: return `${num}rd`;
+    default: return `${num}th`;
+  }
+};
```

Comment:
```suggestion
const getOrdinalSuffix = (num: number): string => {
  const pr = new Intl.PluralRules('en', { type: 'ordinal' });
  const rule = pr.select(num);
  
  switch (rule) {
    case "one": return `${num}st`;
    case "two": return `${num}nd`;
    case "few": return `${num}rd`;
    default: return `${num}th`;
  }
}
```

Consider a JS native approach using Intl.PluralRules